pkgs/top-level/all-packages.nix

```
@@ -17911,6 +17911,8 @@ in
 
     cpupower = callPackage ../os-specific/linux/cpupower { };
 
+    cpupower-gui = callPackage ../os-specific/linux/cpupower/gui.nix { };
```

allow for python packages to be imported
```suggestion
    cpupower-gui = python3Packages.callPackage ../os-specific/linux/cpupower/gui.nix { };
```

---

I don't understand the difference here. If I include this change nix tries to use ninja directly instead of meson and the build fails.

---

that's probably from the setup.py

---

figured it out, there's a `python3Packages.meson` which was getting picked up instead of the build toolchain

---

updated the original suggestion

---

https://gist.github.com/a6a76b53a1a4b33dc857251d5e2cedb3 with `cpupower-gui = callPackage ../os-specific/linux/cpupower/gui.nix { };` works.
https://gist.github.com/ba4aefda50d89ee0413cdc477b0a340d with `cpupower-gui = python3Packages.callPackage ../os-specific/linux/cpupower/gui.nix { };` doesn't, failing with the above error.

---

thanks, that works.